### PR TITLE
tikv-ctl: add command `transfer-leader`.

### PR DESCRIPTION
**Pull request by hicqu**
The command can be run on a crash tikv node to tell pd do a
transfer-leader on a specified region which causes the crash.

### CODE REVIEW

File: src/bin/tikv-ctl.rs [hunk 1]
```diff
@@ -333,16 +344,72 @@ trait DebugExecutor {
...
+                    .expect("Cant get new leader");
+
+                #[derive(Serialize)]
+                struct Body {
+                    name: &'static str,
+                    region_id: u64,
+                    to_store_id: u64,
+                }
+
+                let url = format!("http://{}/pd/api/v1/operators", endpoints[0]);
```
Comment (siddontang) [inline src/bin/tikv-ctl.rs]: seem we should send to PD leader

Comment (hicqu) [reply src/bin/tikv-ctl.rs]: I have tested this, we can send request to followers, and followers will redirect it to leader.

**Comment by siddontang** [comment]
Seem that pd-ctl has already supported this, why do we need it in TiKV?

**Comment by hicqu** [comment]
Because this command can only run on crashed TiKV nodes, which maybe doesn't have `pd-ctl` binary.

**Comment by siddontang** [comment]
Seem the assumption is not suitable. Our deploy tool will guarantee we can use them together.

The only reason I can see here is that we don't use two tools at the same time. 

What is your opinion? /cc @BusyJay

File: src/pd/mod.rs [hunk 2]
```diff
@@ -146,4 +146,13 @@ pub trait PdClient: Send + Sync {
     fn report_split(&self, left: metapb::Region, right: metapb::Region) -> PdFuture<()>;
 }
 
+// PdClient and some other manage commands.
+pub trait PdCtlClient: PdClient {
+    // Get region and its leader by id.
+    fn get_region_and_leader_by_id(
```
Comment (siddontang) [inline src/pd/mod.rs]: I prefer adding a `get_leader` function and we use these separately.

**Comment by BusyJay** [comment]
I think it's OK to use pd-ctl here. If the follower is far behind leader's progress, pd-ctl is still needed to be used any way. But for the long term, maybe it's better to provide only one binary to operate the whole system.

File: src/pd/mod.rs (hunk 2 above)
Comment (hicqu) [reply src/pd/mod.rs]: If we separate them, we must call `GetRegion` API twice, which maybe cause two responses have different content. Maybe it's a problem?

**Comment by hicqu** [comment]
Now `tikv-ctl` call `pd-ctl` directly. PTAL again @siddontang @BusyJay .

**Comment by hicqu** [comment]
It's a bad style. I will examine how to collect all recovery points to one unified style.

**Pull request closed without merging**